src/strands/event_loop/event_loop.py

```diff
@@ -264,8 +278,41 @@ async def event_loop_cycle(agent: "Agent", invocation_state: dict[str, Any]) ->
         yield ForceStopEvent(reason=e)
         logger.exception("cycle failed")
         raise EventLoopException(e, invocation_state["request_state"]) from e
+    finally:
+        if output_schema:
+            handler = StructuredOutputHandler(output_schema)
+            handler.cleanup_all_state(invocation_state)
+
+    # Force structured output tool call if LLM didn't use it automatically
+    if output_schema and stop_reason != "tool_use":
```

Comment:
Given line 238 above, how would we ever get to the point where stop_reason = "tool_use" here?

I *think* that check is redundant

Reply:
I also think we may want to reverse this - I think the only case we would want to do this is if the stop_reason was "end_turn"?